src/Integration/Suppression/ChecksumCalculator.cs

```diff
@@ -0,0 +1,69 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+using System;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace SonarLint.VisualStudio.Integration.Suppression
+{
+    /// <summary>
+    /// Calculates the checksum for an issue using the same method as the SonarQube server
+    /// </summary>
+    /// <remarks>
+    /// For the corresponding code in SL IntelliJ see src\main\java\org\sonarlint\intellij\issue\LiveIssue.java::checksum
+    /// </remarks>
+    internal static class ChecksumCalculator
+    {
+        public static string Calculate(string text)
+        {
+            if (text == null)
+            {
+                throw new ArgumentNullException(nameof(text));
+            }
+
+            string content = Regex.Replace(text, "\\s", ""); // strip whitespace
+            using (MD5 md5Digest = MD5.Create())
```

Comment:
I don't know how expensive creating an instance of _MD5_ each time is, but according to [MSDN](https://msdn.microsoft.com/en-us/library/system.security.cryptography.md5(v=vs.110).aspx) the instance method isn't guaranteed to be thread-safe so the alternative was reuse a single instance but to add locking to this method.